File: JuddFashion.API/JuddFashion.API/Models/Product.cs
namespace JuddFashion.API.Models
{
    public class Product
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public decimal BasePrice { get; set; } 
        public ClothingCategory Category { get; set; }
        public string Brand { get; set; } = string.Empty;
        public string ImageUrl { get; set; } = string.Empty;
        public DateTime DateAdded { get; set; } = DateTime.UtcNow;
        public bool IsActive { get; set; } = true; 
        public ICollection<ProductVariant> Variants { get; set; } = new List<ProductVariant>();
    }
}

File: JuddFashion.CLIENT/src/app/models/product.ts
export interface ProductVariant {
  id: number;
  size: string;
  color: string;
  sku: string;
  stockQuantity: number;
  priceAdjustment: number;
  finalPrice: number;
  inStock: boolean;
}
export interface Product {
  id: number;
  name: string;
  description: string;
  basePrice: number;
  category: string;
  brand: string;
  imageUrl: string;
  dateAdded: string;
  variants: ProductVariant[];
  totalStock: number;
  isAvailable: boolean;
  availableColors: string[];
  availableSizes: string[];
}
